README.rst

```diff
@@ -23,22 +30,304 @@ implementation are in the accepted
+
+to your ``setup.py`` file, then do::
+
+    cmdclassd = register_commands(NAME, VERSION, RELEASE)
+
+where ``NAME`` is the name of your package, ``VERSION`` is the full version
+string, and ``RELEASE`` is a boolean value indicating whether the version is
+a stable released version (``True``) or a developer version (``False``).
+Finally, pass ``cmdclassd`` to the ``setup`` function::
+
+     setup(...,
+           cmdclass=cmdclassd)
+
+The commands we provide or customize are:
+
+**python setup.py test**
+
+This command will automatically build the package, install it to a temporary
+directory, and run the tests using `pytest <http://pytest.org/>`_ on this
+installed version. Note that the bulk of this command is actually defined
+in ``astropy.tests.command.AstropyTest`` - here we simply define the custom
+setuptools command.
+
+**python setup.py sdist**
+
+We redefine ``sdist`` to use the version from distutils rather than from
+setuptools, as the setuptools version requires duplication of information
+in ``MANIFEST.in``.
+
+**python setup.py build_docs**
+
+This command will automatically build the package, then run sphinx to build
+the documentation. Sphinx itself provides a custom setuptools command, which we
+expand with the following options:
+
+* ``-w``: set the return code to 1 if there are any warnings during the build
+  process.
+
+* ``-l``: completely clean previous builds, including files generated by
+  the sphinx-automodapi package (which creates API pages for different
+  functions/classes).
+
+* ``-n``: disable the intersphinx option.
+
+* ``-o``: open the documentation in a browser if a build finishes successfully.
+
+In addition, ``build_docs`` will automatically download and temporarily install
+sphinx-astropy (which is a meta-package that
+provides standardized configuration and documentation dependencies for astropy
+packages) if it isn't already installed. Temporary installation means that the
+package will be installed into an ``.eggs`` directory in the current working
+directory, and it will only be available for the duration of the call to
+``build_docs``.
+
+**python setup.py build_ext**
+
+This is also used when running ``build`` or ``install``. We add several features
+compared to the default ``build_ext`` command:
+
+* For packages with C/Cython extensions, we create a ``packagename._compiler``
+  submodule that contains information about the compilers used.
+
+* Packages that need to build C extensions using the Numpy C API, we allow
+  those packages to define the include path as ``'numpy'`` as opposed to having
+  to import Numpy and call ``get_include``. The goal is to solve the issue that
+  if one has to import Numpy to define extensions, then Numpy has to be
+  installed/available before the package is installed, which means that one
+  needs to install Numpy in a separate installation step.
+
+* We detect broken compilers and replace them with other compilers on-the-fly
```

Comment:
(reminder: remove this if #421 goes in, or remove it *in* #421 if this goes in first)

Reply:
I think this will go in first.